kolibri/plugins/learn/assets/src/views/ExamPage/index.vue

```diff
@@ -5,168 +5,251 @@
+            <KPageContainer>
+
+              <h1 class="number-of-questions">
+                {{ $tr('question', { num: questionNumber + 1, total: exam.question_count }) }}
+              </h1>
+              <ContentRenderer
+                v-if="content && itemId"
+                ref="contentRenderer"
+                :kind="content.kind"
+                :files="content.files"
+                :available="content.available"
+                :extraFields="content.extra_fields"
+                :itemId="itemId"
+                :assessment="true"
+                :allowHints="false"
+                :answerState="currentAttempt.answer"
+                @interaction="saveAnswer"
+              />
+              <ResourceSyncingUiAlert v-else :multiple="false" />
+            </KPageContainer>
+          </main>
+        </KGridItem>
+      </KGrid>
+      <BottomAppBar>
+        <KGrid>
+          <KGridItem
+            :layout12="{ span: 4 }"
+            :layout8="{ span: 2 }"
+            :layout4="{ span: 2 }"
+          >
+            <KButton
+              :disabled="questionNumber === 0"
+              :primary="true"
               :dir="layoutDirReset"
-              class="left-align"
+              :appearanceOverrides="navigationButtonStyle"
+              :class="{ 'left-align': windowIsSmall }"
+              :aria-label="$tr('previousQuestion')"
+              @click="goToQuestion(questionNumber - 1)"
             >
+              <template #icon>
+                <KIcon
+                  icon="back"
+                  :color="$themeTokens.textInverted"
+                  :style="navigationIconStylePrevious"
+                />
+              </template>
+              <span v-if="displayNavigationButtonLabel">{{ $tr('previousQuestion') }}</span>
+            </KButton>
+
+          </KGridItem>
+
+          <KGridItem
+            :layout12="{ span: 4 }"
+            :layout8="{ span: 2 }"
+            :layout4="{ span: 1 }"
+          >
+            <div class="" style="text-align:center">
               <div v-if="!missingResources" class="answered">
+
                 {{ answeredText }}
-              </div>
-              <KButton
-                v-if="!missingResources"
-                :text="$tr('submitExam')"
-                :primary="false"
-                appearance="flat-button"
-                @click="toggleModal"
-              />
-              <div v-if="missingResources" class="nosubmit">
-                {{ $tr('unableToSubmit') }}
+                <div v-if="missingResources" class="nosubmit">
+                  {{ $tr('unableToSubmit') }}
+                </div>
               </div>
             </div>
+          </KGridItem>
+
+
+          <KGridItem
+            :layout12="{ span: 4 }"
+            :layout8="{ span: 4 }"
+            :layout4="{ span: 2 }"
+          >
+            <div class="" style="float:right">
+              <div class="" style="float:right">
```

Comment:
I'm wondering why the two floats are needed here. 

In this case I think that we could leverage some other CSS properties for a cleaner and more consistent alignment & layout.

By using the `position: relative;` property on the parent element (the outermost `<div>` here), we can then use `position: absolute;` on any child of that div so that you can position that child w/ the `right/left/top/bottom` CSS properties

In this case, I'd suggest trying the following structure -- which may not be exactly right but should hopefully demonstrate the idea:

```
<KGridItem ...>
    <div style="position: relative;"> 
        <KButton style="position: absolute; right: 0">
```

This way it can be reasoned about kinda like:

_The `KButton` is _absolutely_ 0px from the right edge, _relative_ to the parent `div`_